tests/p2p/test_peer_id.py

```diff
@@ -17,7 +19,7 @@
 class PeerIdTest(unittest.TestCase):
     def test_invalid_id(self) -> None:
         p1 = Peer()
-        p1.id = not_none(p1.id)[::-1]
+        p1.id = PeerId(str(not_none(p1.id))[::-1])
```

Comment:
Done, I've fixed all the cases that match `rg -t py 'not_none\(.*\.id\)'`.

Reply:
Actually, I won't include this change. Since `Peer.id` is `Optional[PeerId]` this still makes sense. I realized when mypy warned for other cases, it doesn't warn for `str(peer.id)` because `str(None)` is valid. I'll do this in #1115 instead, since it changes `Peer.id` to `PeerId` without the `Optional`.

Reply:
Ok. Thanks!